src/malva/java/text/DecimalFormatTest.java

```diff
@@ -19,7 +23,49 @@ public static void testFormat() {
     assertEquals("-4,3 unit", df.format(-4.3));
   }
 
+  public static void testFormatToCharacterIterator() {
+    final DecimalFormat df = new DecimalFormat("0.##");
+
+    // Check first for throwables
+    assertThrows(new Block() {
+      @Override public void run() {
+        df.formatToCharacterIterator(null);
+      }
+    }, NullPointerException.class);
+
+    assertThrows(new Block() {
+      @Override public void run() {
+        df.formatToCharacterIterator(new Object());
+      }
+    }, IllegalArgumentException.class);
+
+    // Attributes check
+    Set<?> attrs;
+    attrs = formatAndGetAttrs(df, 0);
+    assertTrue(attrs.containsAll(Arrays.asList(Field.INTEGER)));
+
+    attrs = formatAndGetAttrs(df, Double.valueOf(1.23));
+    assertTrue(attrs.containsAll(Arrays.asList(Field.INTEGER, Field.DECIMAL_SEPARATOR, Field.FRACTION)));
+
+    attrs = formatAndGetAttrs(df, Double.valueOf(-1.23));
+    assertTrue(attrs.containsAll(Arrays.asList(Field.SIGN, Field.INTEGER, Field.DECIMAL_SEPARATOR, Field.FRACTION)));
+
+    attrs = formatAndGetAttrs(df, new BigDecimal(Math.PI));
+    assertTrue(attrs.containsAll(Arrays.asList(Field.INTEGER, Field.DECIMAL_SEPARATOR, Field.FRACTION)));
+
+    attrs = formatAndGetAttrs(df, new BigDecimal("123"));
+    assertTrue(attrs.containsAll(Arrays.asList(Field.INTEGER)));
+
+    attrs = formatAndGetAttrs(df, Long.valueOf("12345"));
+    assertTrue(attrs.containsAll(Arrays.asList(Field.INTEGER)));
+  }
+
+  private static Set<?> formatAndGetAttrs(DecimalFormat df, Object obj) {
```

Comment:
Instead of using `Set<?>`, why not specify the class of the elements explicitly?

Reply:
My thought process was that this does not really matter since I just want to use `containsAll(Collection<?>)` with the returned Set. I'll add it anyways for better readability.